src/generators/javascript/JavaScriptGenerator.ts

```diff
@@ -27,9 +31,29 @@ export class JavaScriptGenerator extends AbstractGenerator<JavaScriptOptions> {
   ) {
     super('JavaScript', JavaScriptGenerator.defaultOptions, options);
   }
+  
+  /**
+   * Render a complete model result where the model code, library and model dependencies are all bundled appropriately.
+   *
+   * @param model
+   * @param inputModel
+   * @param options
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, options: JavaScriptRenderCompleteModelOptions): Promise<RenderOutput> {
+    const outputModel = await this.render(model, inputModel);
+    const modelDependencies = model.getNearestDependencies().map((dependencyModelName) => {
+      const formattedDependencyModelName = this.options.namingConvention?.type ? this.options.namingConvention.type(dependencyModelName, { inputModel, model: inputModel.models[String(dependencyModelName)] }) : dependencyModelName;
+      return `const ${formattedDependencyModelName} = require('./${formattedDependencyModelName}');`;
```

Comment:
Have in mind that someone will want to render models with ESM not CJS module system, so you `JavaScriptRenderCompleteModelOptions` should have option to choose it - by default we can use ESM.

Reply:
Good catch! Adapting.